core/src/main/java/org/apache/iceberg/BaseTable.java

```diff
@@ -29,8 +30,12 @@
  * Base {@link Table} implementation.
  * <p>
  * This can be extended by providing a {@link TableOperations} to the constructor.
+ * <p>
+ * Serializing and deserializing a BaseTable object returns a read only implementation of the BaseTable using a
+ * {@link StaticTableOperations}. This way no Catalog related calls are needed when reading the table data after
+ * deserialization.
  */
-public class BaseTable implements Table, HasTableOperations {
+public class BaseTable implements Table, HasTableOperations, Serializable {
```

Comment:
I'm also wondering if we should make `Table` extend `Serializable`. Not something to decide now, but I think this is a reasonable way to make that work.